### PR TITLE
Merging Rust STD side of std-fs integration

### PR DESCRIPTION
**Pull request by CPTforever**
I've added some code that will interface with the Twizzler runtime. This allows user programs to use the Rust std fs interface to access Twizzler objects.

### CODE REVIEW

File: library/std/src/sys/hermit/fd.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ impl FileDesc {
     }
 
     pub fn write(&self, buf: &[u8]) -> io::Result<usize> {
-        let result = cvt(unsafe { abi::write(self.fd.as_raw_fd(), buf.as_ptr(), buf.len()) })?;
+        let result = cvt(unsafe { abi::write(self.fd.as_raw_fd(), buf.as_mut_ptr(), buf.len()) })?;
```
Comment (dbittman) [inline library/std/src/sys/hermit/fd.rs]: Why is there a change in the hermit code?

File: library/std/src/sys/twizzler/fs.rs [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+    }
+}
+
+impl File {
+    pub fn open(path: &Path, opts: &OpenOptions) -> io::Result<File> {
+        run_path_with_cstr(path, |path| File::open_c(&path, opts))
+    }
+
+    pub fn open_c(path: &CStr, opts: &OpenOptions) -> io::Result<File> {
+        println!("opening 1");
```
Comment (dbittman) [inline library/std/src/sys/twizzler/fs.rs]: We should try to keep the code free of these indicators.

File: library/std/src/sys/twizzler/fs.rs [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+    }
+
+    pub fn set_times(&self, _times: FileTimes) -> io::Result<()> {
+        Err(Error::from_raw_os_error(22))
+    }
+}
+
+impl DirBuilder {
+    pub fn new() -> DirBuilder {
+        println!("Inited DirBuilder! :)");
```
Comment (dbittman) [inline library/std/src/sys/twizzler/fs.rs]: Here, too

File: library/std/src/sys/twizzler/fs.rs [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+    }
+
+    pub fn mkdir(&self, _p: &Path) -> io::Result<()> {
+        unsupported()
+    }
+}
+
+impl fmt::Debug for File {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "No")
```
Comment (dbittman) [inline library/std/src/sys/twizzler/fs.rs]: This should do something better -- write the fd number or something?

File: library/std/src/sys/twizzler/mod.rs [hunk 5]
```diff
@@ -6,7 +6,8 @@ pub mod args;
 #[path = "../unix/cmath.rs"]
 pub mod cmath;
 pub mod env;
-#[path = "../unsupported/fs.rs"]
+//#[path = "../unsupported/fs.rs"]
```
Comment (dbittman) [inline library/std/src/sys/twizzler/mod.rs]: Remove the commented #[attr]

**Review by dbittman — CHANGES_REQUESTED** [review]
Just some small changes

**Comment by CPTforever** [comment]
Undid change to hermit 
Removed all print statements
Removed commented $[attr]
Added a display trait to FileDesc and a debug trait to FIle so it should print out the file descriptor number

**Pull request merged**